runelite-client/src/main/java/net/runelite/client/plugins/timers/Timers.java

```diff
@@ -0,0 +1,170 @@
+	private final TimersOverlay overlay = new TimersOverlay(this);
+
+	@Override
+	public Overlay getOverlay()
+	{
+		return overlay;
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		// Initialize overlay config
+		overlay.updateConfig();
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+
+	}
+
+	public TimersConfig getConfig()
+	{
+		return config;
+	}
+
+	@Subscribe
+	public void updateConfig(ConfigChanged event)
+	{
+		overlay.updateConfig();
+	}
+
+	@Subscribe
+	public void onChatMessage(ChatMessage event)
+	{
+		if (event.getType() == ChatMessageType.FILTERED)
+		{
+			if (event.getMessage().equals("You drink some of your stamina potion.") && config.showStamina())
+			{
+				createGameTimer(STAMINA);
+			}
+
+			if (event.getMessage().equals("<col=8f4808>Your stamina potion has expired.</col>"))
+			{
+				removeGameTimer(STAMINA);
+			}
+
+			if (event.getMessage().equals("You drink some of your antifire potion.") && config.showAntiFire())
+			{
+				createGameTimer(ANTIFIRE);
+			}
+
+			if (event.getMessage().equals("You drink some of your extended antifire potion.") && config.showExAntiFire())
+			{
+				createGameTimer(EXANTIFIRE);
+			}
+
+			if (event.getMessage().equals("<col=7f007f>Your antifire potion has expired.</col>"))
+			{
+				//they have the same expired message
+				removeGameTimer(ANTIFIRE);
+				removeGameTimer(EXANTIFIRE);
+			}
+
+			if (event.getMessage().contains("You drink some of your overload potion") && config.showOverload())
+			{
+				createGameTimer(OVERLOAD);
+			}
+
+			if ((event.getMessage().equals("You add the furnace.") || event.getMessage().contains("You repair your cannon, restoring it to working order.")) && config.showCannon())
+			{
+				createGameTimer(CANNON);
+			}
+
+			if (event.getMessage().equals("You pick up the cannon. It's really heavy."))
+			{
+				removeGameTimer(CANNON);
+			}
+
+			if (event.getMessage().contains("You drink some of your super antivenom potion") && config.showAntiVenom())
+			{
+				createGameTimer(SUPERANTIVENOM);
+			}
+
+			if (event.getMessage().equals("You are charged to combine runes!") && config.showMagicImbue())
+			{
+				createGameTimer(MAGICIMBUE);
+			}
+		}
+
+		if ( event.getType() == ChatMessageType.SERVER)
+		{
+			//why would they split magic imbue game messages types?
+			if (event.getMessage().equals("Your Magic Imbue charge has ended."))
+			{
+				removeGameTimer(MAGICIMBUE);
+			}
+			if (event.getMessage().equals("<col=4f006f>A teleblock spell has been cast on you. It will expire in 5 minutes, 0 seconds.</col>") && config.showTeleblock())
+			{
+				createGameTimer(FULLTB);
+			}
+
+			if (event.getMessage().equals("<col=4f006f>A teleblock spell has been cast on you. It will expire in 2 minutes, 30 seconds.</col>") && config.showTeleblock())
+			{
+				createGameTimer(HALFTB);
+			}
+		}
+	}
+
+	public void createGameTimer(GameTimers timer)
+	{
+		removeGameTimer(timer);
+
+		timer.setTime(Instant.now());
```

Comment:
I probably wouldn't mix what is defining the timers (GameTimers) with what is keeping track of running timers.